packages/form-js-viewer/src/render/hooks/useRestrictedMultiSelectValues.js

```diff
@@ -0,0 +1,32 @@
+import { useEffect } from 'preact/hooks';
+import { LOAD_STATES } from './useOptionsAsync';
+
+export default function(props) {
+
+  const {
+    field,
+    options,
+    loadState,
+    onChange,
+    values
+  } = props;
+
+  // Ensures that the values are always a subset of the options
+  useEffect(() => {
+
+    if (loadState !== LOAD_STATES.LOADED) {
+      return;
+    }
+
+    const hasValuesNotInOptions = values.some(v => !options.map(o => o.value).includes(v));
+
+    if (hasValuesNotInOptions) {
+      onChange({
+        field,
+        value: values.filter(v => options.map(o => o.value).includes(v))
```

Comment:
As in the comment above having updates here might be unexpected. What would be the disadvantage of only sanitizing `value` here and not updating it directly? I guess it would lead to unexpected side effects as we never really clean things up?

Reply:
Yeah because then you have a mismatch between the form and the actual state. So if you have lets say, a feel expression dependent on this value, it'll still evaluate despite the form not showing that value being selected.